core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -31,39 +31,45 @@ class OffsetCheckpoint(val file: File) extends Logging {
 
   def write(offsets: Map[TopicAndPartition, Long]) {
     lock synchronized {
-      // write to temp file and then swap with the existing file
-      val temp = new File(file.getAbsolutePath + ".tmp")
-
-      val fileOutputStream = new FileOutputStream(temp)
-      val writer = new BufferedWriter(new OutputStreamWriter(fileOutputStream))
       try {
-        // write the current version
-        writer.write(0.toString)
-        writer.newLine()
-      
-        // write the number of entries
-        writer.write(offsets.size.toString)
-        writer.newLine()
+        // write to temp file and then swap with the existing file
+        val temp = new File(file.getAbsolutePath + ".tmp")
+
+        val fileOutputStream = new FileOutputStream(temp)
+        val writer = new BufferedWriter(new OutputStreamWriter(fileOutputStream))
+        try {
+          // write the current version
+          writer.write(0.toString)
+          writer.newLine()
 
-        // write the entries
-        offsets.foreach { case (topicPart, offset) =>
-          writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
+          // write the number of entries
+          writer.write(offsets.size.toString)
           writer.newLine()
+
+          // write the entries
+          offsets.foreach { case (topicPart, offset) =>
+            writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
+            writer.newLine()
+          }
+
+          // flush the buffer and then fsync the underlying file
+          writer.flush()
+          fileOutputStream.getFD().sync()
+        } finally {
+          writer.close()
         }
-      
-        // flush the buffer and then fsync the underlying file
-        writer.flush()
-        fileOutputStream.getFD().sync()
-      } finally {
-        writer.close()
-      }
-      
-      // swap new offset checkpoint file with previous one
-      if(!temp.renameTo(file)) {
-        // renameTo() fails on Windows if the destination file exists.
-        file.delete()
-        if(!temp.renameTo(file))
-          throw new IOException("File rename from %s to %s failed.".format(temp.getAbsolutePath, file.getAbsolutePath))
+
+        // swap new offset checkpoint file with previous one
+        if(!temp.renameTo(file)) {
+          // renameTo() fails on Windows if the destination file exists.
+          file.delete()
+          if(!temp.renameTo(file))
+            throw new IOException("File rename from %s to %s failed.".format(temp.getAbsolutePath, file.getAbsolutePath))
+        }
+      } catch {
+        case e: FileNotFoundException =>
```

Comment:
@MayureshGharat I feel like its better we catch this in KafkaServer itself and call a halt. It will be a single place to catch this exception and as you said its multiple places like KafkaApis etc.. .

For EROFS , my initial thought was to string match the exception I don't see an alternate way other than that.